airbyte-integrations/connectors/source-googleanalytics-singer/source_googleanalytics_singer/source.py

```diff
@@ -44,14 +47,38 @@ def configure(self, raw_config: json, temp_dir: str) -> json:
         with open(credentials, "w") as fh:
             fh.write(raw_config["credentials_json"])
         raw_config["key_file_location"] = credentials
+        if raw_config.get("custom_reports", "").strip() and json.loads(raw_config["custom_reports"]):
```

Comment:
If the field input in "custom_reports" is a non-empty string, it must be valid JSON or the validation should fail. See comment below about throwing errors -- we'll need to move this validation into the check method.